peps/pep-0782.rst

```diff
@@ -0,0 +1,371 @@
+            goto error;
+        }
+        return PyBytesWriter_Finish(writer);
+
+    error:
+        PyBytesWriter_Discard(writer);
+        return NULL;
+    }
+
+
+Create the bytes string "abc"
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Example creating the string ``"abc"``, with a fixed size of 3 bytes::
+
+    PyObject* create_abc(void)
+    {
+        PyBytesWriter *writer = PyBytesWriter_Create(3);
+        if (writer == NULL) {
+            return NULL;
+        }
+
+        char *str = PyBytesWriter_GetData(writer);
+        memcpy(str, "abc", 3);
+        return PyBytesWriter_Finish(writer);
+    }
+
+GrowAndUpdatePointer() example
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Example using a pointer to write bytes and to track the written size.
+
+Create the string ``"Hello World"``::
+
+    PyObject* grow_example(void)
+    {
+        // Allocate 10 bytes
+        PyBytesWriter *writer = PyBytesWriter_Create(10);
+        if (writer == NULL) {
+            return NULL;
+        }
+
+        // Write some bytes
+        char *buf = PyBytesWriter_GetData(writer);
+        memcpy(buf, "Hello ", strlen("Hello "));
+        buf += strlen("Hello ");
+
+        // Allocate 10 more bytes
+        buf = PyBytesWriter_GrowAndUpdatePointer(writer, 10, buf);
+        if (buf == NULL) {
+            PyBytesWriter_Discard(writer);
+            return NULL;
+        }
+
+        // Write more bytes
+        memcpy(buf, "World", strlen("World"));
+        buf += strlen("World");
+
+        // Truncate the string at 'buf' position
+        // and create a bytes object
+        return PyBytesWriter_FinishWithPointer(writer, buf);
+    }
+
+
+Implementation
+==============
+
+* `Pull request gh-131681 <https://github.com/python/cpython/pull/131681>`__.
+
+
+Backwards Compatibility
+=======================
+
+There is no impact on the backward compatibility, only new APIs are
+added.
+
+
+Projects using _PyBytes_Resize()
+================================
+
+A code search on PyPI top 8,000 projects finds 41 projects using
+``_PyBytes_Resize``:
+
+* Nuitka (2.6)
+* PyBluez (0.23)
+* PyICU (2.14)
+* PyICU-binary (2.7.4)
+* SimpleParse (2.2.4)
+* apsw (3.48.0.0)
+* asyncio (3.4.3)
+* billiard (4.2.1)
+* bitarray (3.0.0)
+* blosc (1.11.2)
+* casadi (3.6.7)
+* catboost (1.2.7)
+* cython (3.0.11)
+* ddtrace (2.20.0)
+* deflate (0.7.0)
+* isal (1.7.1)
+* m2crypto (0.43.0)
+* msgspec (0.19.0)
+* multiprocess (0.70.17)
+* mysql-connector (2.2.9)
+* mysql-connector-python-rf (2.2.2)
+* mysqlclient (2.2.7)
+* orjson (3.10.15)
+* ormsgpack (1.7.0)
+* pickle5 (0.0.12)
+* pillow (11.1.0)
+* psycopg2 (2.9.10)
+* psycopg2-binary (2.9.10)
+* pyarrow (19.0.0)
+* pybase64 (1.4.0)
+* pygobject (3.50.0)
+* pygresql (6.1.0)
+* pyobjc_core (11.0)
+* pysam (0.22.1)
+* pyzstd (0.16.2)
+* rcssmin (1.2.0)
+* rjsmin (1.2.3)
+* zipfile-deflate64 (0.2.0)
+* zlib_ng (0.5.1)
+* zodbpickle (4.1.1)
+* zstandard (0.23.0)
+
+
+Discussions
+===========
```

Comment:
Again I'm not sure what this section is for -- the list of prior discussions should probably be contextualised in the _Motivation_ or _Rationale_ sections, with the differences from this new PEP/proposal.
```suggestion
Prior Discussions
=================
```